sklearn/mixture/tests/test_bayesian_mixture.py

```diff
@@ -0,0 +1,365 @@
+                         "The parameter 'spherical covariance_prior' "
+                         "should be greater than 0., but got %.3f."
+                         % bad_covariance_prior_,
+                         bgmm.fit, X)
+
+    # Check correct init for the default value of covariance_prior
+    covariance_prior_default = {
+        'full': np.atleast_2d(np.cov(X.T)),
+        'tied': np.atleast_2d(np.cov(X.T)),
+        'diag': np.var(X, axis=0, ddof=1),
+        'spherical': np.var(X, axis=0, ddof=1).mean()}
+
+    bgmm = BayesianGaussianMixture(random_state=0)
+    for cov_type in ['full', 'tied', 'diag', 'spherical']:
+        bgmm.covariance_type = cov_type
+        bgmm.fit(X)
+        assert_almost_equal(covariance_prior_default[cov_type],
+                            bgmm.covariance_prior_)
+
+
+def test_bayesian_mixture_check_is_fitted():
+    rng = np.random.RandomState(0)
+    n_samples, n_features = 10, 2
+
+    # Check raise message
+    bgmm = BayesianGaussianMixture(random_state=rng)
+    X = rng.rand(n_samples, n_features)
+    assert_raise_message(ValueError,
+                         'This BayesianGaussianMixture instance is not '
+                         'fitted yet.', bgmm.score, X)
+
+
+def test_bayesian_mixture_weights():
+    rng = np.random.RandomState(0)
+    n_samples, n_features = 10, 2
+
+    X = rng.rand(n_samples, n_features)
+    bgmm = BayesianGaussianMixture(random_state=rng).fit(X)
+
+    # Check the weights values
+    expected_weights = (bgmm.dirichlet_concentration_ /
+                        np.sum(bgmm.dirichlet_concentration_))
+    predected_weights = bgmm.weights_
+
+    assert_almost_equal(expected_weights, predected_weights)
+
+    # Check the weights sum = 1
+    assert_almost_equal(np.sum(bgmm.weights_), 1.0)
+
+
+def test_monotonic_likelihood():
+    # We check that each step of the EM without regularization improve
+    # monotonically the training set of the bound
+    rng = np.random.RandomState(0)
+    rand_data = RandomData(rng, scale=7)
+    n_components = rand_data.n_components
+
+    for covar_type in COVARIANCE_TYPE:
+        X = rand_data.X[covar_type]
+        bgmm = BayesianGaussianMixture(n_components=2 * n_components,
+                                       covariance_type=covar_type,
+                                       warm_start=True, max_iter=1,
+                                       random_state=rng, tol=1e-4)
+        current_lower_bound = -np.infty
+        with warnings.catch_warnings():
+            warnings.simplefilter("ignore", ConvergenceWarning)
+            # Do one training iteration at a time so we can make sure that the
+            # training log likelihood increases after each iteration.
+            for _ in range(500):
+                prev_lower_bound = current_lower_bound
+                try:
+                    current_lower_bound = bgmm.fit(X).lower_bound_
+                except ConvergenceWarning:
+                    pass
```

Comment:
ConvergenceWarning is a warning, not an exception. You can remove that try except construct. And I did not get any warning on stderr when doing so.